Bug 1370594: New page headers, new zone headers, nav tweaks

stephaniehobson opened the pull request:
I ended up having to make changes to the zones to support the nav and page header changes. This was planned eventually anyway, it just happened a bit sooner. 

**Nav changes:**
- Move blue border to top of global nav
  - Add box shadow under it

**Zone changes:**
- Make zone landing pages match other article landing pages (mostly)
  - Remove breadcrumbs from zone root pages
- Remove zone specific article footers
- Add new zone icons
- Remove zone coloured background
  - Update code which relied on there being a background
- Update associated tests

**Page changes:**
- Created new document_head block
  - Apply to article view/edit/move/translate/create pages
  - Add blue background to document head
  - Decreased mobile title size when inside .document-head
- Line up buttons and crumb horizontally
- Tweak mobile and tablet display
- Styles apply to non-wiki pages to moved styles to main.css

Testing notes:
- This change affects wiki pages, wiki landing pages, zone pages, zone landing pages, edit, move, translate, and create pages, and the revision dashboard (and should not affect other pages)
-  Tagging @escattone to review the test updates

stephaniehobson commented:
@schalkneethling I can submit just the zone commit separately for review if that would make it easier but the nav and header updates rely on that work to be done first.

schalkneethling commented:
There is currently a flake8 error present:
https://travis-ci.org/mozilla/kuma/jobs/260413234#L559

schalkneethling commented:
This might be by design but, the book icon on `/Learn` is coloured and sized differently than on `Getting_started_with_the_web`. Thought I would mention it in case it is not intentional

<img width="1468" alt="screen shot 2017-08-03 at 16 18 10" src="https://user-images.githubusercontent.com/10350960/28926310-aecc17f4-7867-11e7-9889-5678e13ae855.png">

<img width="1486" alt="screen shot 2017-08-03 at 16 17 51" src="https://user-images.githubusercontent.com/10350960/28926318-b8060dac-7867-11e7-82a5-1df0cd23e0ee.png">

schalkneethling commented:
Also, probably by design, but the `box-shadow` below the nav bar looks a little odd on the right hand side on large viewports. Feels like the "Sign in" widget should be aligned all the way to the right of the viewport.

<img width="249" alt="screen shot 2017-08-03 at 15 48 28" src="https://user-images.githubusercontent.com/10350960/28926460-26021ada-7868-11e7-8086-cc6de40f804e.png">

schalkneethling commented:
Curious, is the background colour for the pager numbers correct?

<img width="143" alt="screen shot 2017-08-03 at 16 29 47" src="https://user-images.githubusercontent.com/10350960/28927478-f0d135e6-786a-11e7-91bc-a553c225ff1b.png">

schalkneethling commented:
On the revision dashboard, when comparing to edits, the copy starts to overlap on medium(tablet) sized viewport sizes:

<img width="541" alt="screen shot 2017-08-03 at 16 31 21" src="https://user-images.githubusercontent.com/10350960/28927635-5f6418ca-786b-11e7-9f9c-3f4e9da810da.png">

Perhaps they should be stacked at this width?

schalkneethling commented:
On small viewports (roughly large mobile), the header layout breaks a bit.

<img width="476" alt="screen shot 2017-08-03 at 16 37 26" src="https://user-images.githubusercontent.com/10350960/28927690-7ffc3a86-786b-11e7-8deb-34ac7c4eaa37.png">

schalkneethling commented:
Also, at 480px - 769px the icon position if offset:

<img width="424" alt="screen shot 2017-08-03 at 16 26 41" src="https://user-images.githubusercontent.com/10350960/28927756-a881e08c-786b-11e7-8bdf-a5e46633202e.png">

schalkneethling commented:
When the content is laid out rtl and on a mobile viewport, the `x` overlaps the top navigation item

<img width="119" alt="screen shot 2017-08-03 at 18 40 06" src="https://user-images.githubusercontent.com/10350960/28932893-5de02f7e-787b-11e7-9cb1-737dc41c1fe4.png">

schalkneethling commented:
Probably not related to this PR, and might be a known issue but, with rtl languages the translation edit pages kinda falls apart when the viewport is resized. Even at 1200px, especially the editor section body, and header sizes independently.

<img width="1180" alt="screen shot 2017-08-03 at 18 46 37" src="https://user-images.githubusercontent.com/10350960/28933245-b4badae6-787c-11e7-8a39-c56b1e008188.png">

schalkneethling commented:
In IE11 the three grey blocks are not getting centred at smaller viewports. This causes the right side of the the last block to be cut of, and no horizontal scroll bars are added.

![virtualbox_windows 8 1 - ie11_03_08_2017_19_02_27](https://user-images.githubusercontent.com/10350960/28933760-9d115cb0-787e-11e7-8f17-99d4a0038dc4.png)

stephaniehobson commented:
Large grey zone icons
The zone icon is larger and darker on the zone landing page. On the sub pages clicking the icon will take you to the landing page, on the landing page there's no need to click it. I've edited the HTML to remove the link altogether now.

Box shadow
I've updated the homepage background to span the width of the screen now, like the article header backgrounds do. I think this counters the weirdness of the black in the nav bar a bit. This let me get rid of the negative margin hack on that element, so yay!

Page numbers
Replaced another hard coded colour with a variable name 🎉 

Overlapping comparison
Probably needs to be addressed elsewhere.

Mobile layout of article header
That is displaying as intended but I'm concerned that it looks broken to you 😞 I struggled a bit with this breakpoint because the spacing looks off unless both the parent article and the current article have long titles. Would left aligning things make it look more intentional?

Icon position on tablet
Fixed by making it the correct size.

Close button in submenu in RTL
Fixed.

Edit pages
Needs to be addressed separately.

Grey boxes on HTML page
Needs to be addressed separately.  Seems like a IE/flexbox bug 😣

schalkneethling commented on kuma/static/styles/includes/_mixins.scss:
```diff
@@ -365,7 +364,7 @@ These are not dynamic but serve as mixins
     @include set-font-size(22px);
     border: 5px solid $accent-light;
     border-width: 5px 0 5px 0;
```
Nit: Could be shortened to `border-width: 5px 0;`?

schalkneethling commented on kuma/static/styles/components/structure/main-header.scss:
```diff
@@ -4,13 +4,17 @@ Main header, wraps all components of logo, main, and secondary navigation
+    z-index: 1;
     padding-top: $mobile-center-spacing;
+    border: 5px solid $accent-light;
+    border-width: 5px 0 0 0;
```
Nit: Could be shortened to `border-width: 5px 0 0;`?

schalkneethling commented:
> Would left aligning things make it look more intentional?

Would definitely work better I believe.

schalkneethling commented:
Other than the last couple of small items, this is looking great r+w/c ☕️

stephaniehobson commented:
Nits addressed. 

Looked at moving language switcher to left on mobile but it required too much fighting with some global styles. (This part of the code needs to be refactored, too many un-related components are using the same class). Over all this is still an improvement for mobile.
